package/MDAnalysis/core/topologyattrs.py

```diff
@@ -2629,6 +2629,86 @@ class Bonds(_Connection):
     transplants = defaultdict(list)
     _n_atoms = 2
 
+    def __init__(self, values, types=None, guessed=False, order=None):
+        """
+        Initialise a Bond Attribute including a BondTable
+
+        .. versionadded:: 2.3.0
+        """
+        super().__init__(values, types, guessed, order)
+        self._bondtable = BondTable(values, self.types, self._guessed,
+                                    self.order)
+
+    def get_atoms(self, ag):
+        """
+        Get Bond values where the atom indices are in the given atomgroup by
+        querying the BondTable
+
+        Parameters
+        ----------
+        ag : AtomGroup
+
+        .. versionadded:: 2.3.0
+        """
+        b_idx, types, guessed, order = self._bondtable.get_b_t_g_o_slice(ag.ix)
+        return TopologyGroup(b_idx, ag.universe,
```

Comment:
This line reinitialized the Topology group every time get_atoms is called. Shouldn't this be cached?